### PR TITLE
Remove triage/needs-triage label after closing issue

### PR DESCRIPTION
**Pull request by saumya1singh**
This is PR for #30 :)
Hopefully, now I understood the correct flow!

CC @FroMage @gsmet

### CODE REVIEW

File: src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import io.quarkus.bot.util.Labels;
+import org.jboss.logging.Logger;
+import org.kohsuke.github.GHEventPayload;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHLabel;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+public class RemoveLabelFromClosedIssue {
```
Comment (saumya1singh) [inline src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java]: @gsmet 
Let me know if you want this method to be inside the `TriageIssue` class :)

Comment (gsmet) [reply src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java]: No, it's perfect, as explained elsewhere, the idea is to have a class per rule.

File: src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+   private static final Logger LOG = Logger.getLogger(RemoveLabelFromClosedIssue.class);
+
+   @Inject
+   QuarkusBotConfig quarkusBotConfig;
```
Comment (gsmet) [inline src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java]: You can remove the config file here (and all the code related to it), we don't need it.

File: src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+   void onClose(@Issue.Closed GHEventPayload.Issue issuePayload,
```
Comment (gsmet) [inline src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java]: Yes, that's the idea, looks like you understood the event based architecture, nice!

File: src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+      GHIssue issue = issuePayload.getIssue();
+      for (GHLabel label : issue.getLabels()) {
+         if (label.getName().equals(Labels.TRIAGE_NEEDS_TRIAGE)){
+            if (!quarkusBotConfig.isDryRun()) {
+               // issue.comment("TEST CLOSE");
+               issue.removeLabels(Labels.TRIAGE_NEEDS_TRIAGE);
+            } else {
+               LOG.info("Issue #" + issue.getNumber() + " - Remove label: " + String.join(", ", label.getName()));
+            }
```
Comment (gsmet) [inline src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java]: You can add a `break;` here. As soon as you have found the `triage/needs-rebase` label, you don't need to do anything else.

It's not really mandatory here but it's good practice!

File: src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      if (quarkusBotConfigFile == null) {
+         LOG.error("Unable to find triage configuration.");
+         return;
+      }
+
+      GHIssue issue = issuePayload.getIssue();
+      for (GHLabel label : issue.getLabels()) {
+         if (label.getName().equals(Labels.TRIAGE_NEEDS_TRIAGE)){
+            if (!quarkusBotConfig.isDryRun()) {
+               // issue.comment("TEST CLOSE");
```
Comment (gsmet) [inline src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java]: This should be removed!

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks like you understood how things work now, nice!

I added some minor comments. Could you have a look when you have some time?

Thanks!

**Comment by saumya1singh** [comment]
> Looks like you understood how things work now, nice!
> 
> I added some minor comments. Could you have a look when you have some time?
> 
> Thanks!

Sure, Done :)

File: src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java (hunk 5 above)
Comment (saumya1singh) [reply src/main/java/io/quarkus/bot/RemoveLabelFromClosedIssue.java]: Done

**Review by gsmet — APPROVED** [review]
I made some small adjusments before merging. I will let you have a look at the final result.

Thanks!

**Comment by gsmet** [comment]
And merged, thanks!

**Pull request merged**